File: webappd/src/main/java/webappd/Broadcaster.java
package webappd;

import java.util.Set;
import java.util.concurrent.CopyOnWriteArraySet;

class Broadcaster {
    private static final Broadcaster INSTANCE = new Broadcaster();

    private Set<Socket> sockets = new CopyOnWriteArraySet<>();

    static Broadcaster getInstance() {
        return INSTANCE;
    }

    void addSocket(Socket socket) {
        this.sockets.add(socket);
    }

    void removeSocket(Socket socket) {
        this.sockets.remove(socket);
    }

    void broadcast(String txt) {
        this.sockets.forEach(socket -> socket.send(txt));
    }
}

File: webappd/src/main/java/webappd/Broadcaster.kt
package webappd

import java.util.concurrent.CopyOnWriteArraySet

internal class Broadcaster {

    private val sockets = CopyOnWriteArraySet<Socket>()

    fun addSocket(socket: Socket) {
        this.sockets.add(socket)
    }

    fun removeSocket(socket: Socket) {
        this.sockets.remove(socket)
    }

    fun broadcast(txt: String) {
        this.sockets.forEach { socket -> socket.send(txt) }
    }

    companion object {
        val instance = Broadcaster()
    }
}
